### PR TITLE
refactor: Convert TicketHolder Model Def to dataclass

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Part of #6465 

#### Short description of what this resolves:
Added Initial Data Class Definition for TicketHolder. Will Proceed with the other models if this is OK to go.

#### Changes proposed in this pull request:

- Moved to Data Classes scheme for `TicketHolder`
- Upgraded Python version to 3.7 in Travis Config file

### CODE REVIEW

**Comment by iamareebjamal** [comment]
https://stackoverflow.com/questions/55577165/sqlalchemy-with-dataclass-default-not-populating-postgres-database

**Comment by mrsaicharan1** [comment]
@iamareebjamal I'm just a bit unsure about the type hinting for the relationships defined in the class. I'll test it out by purchasing tickets on FE and see if it breaks mid-way.

**Comment by iamareebjamal** [comment]
Type hinting can never break anything because it is ignored at runtime. If something gets broken, it might be due to dataclass

**Comment by iamareebjamal** [comment]
You need to update the python version to 3.7 in travis CI

**Comment by mrsaicharan1** [comment]
Tested this for a normal ticket purchase. Works fine.

**Comment by mrsaicharan1** [comment]
Python 3.7 couldn't be configured by changing just the version number. Had to specify the xenial source explicitly.
https://travis-ci.community/t/unable-to-download-python-3-7-archive-on-travis-ci/639/10

File: .travis.yml [hunk 1]
```diff
@@ -9,9 +9,11 @@ addons:
...
       - libssl-dev
       - libffi-dev
-
-python:
-  - '3.6'
+matrix:
+    include:
+        - python: 3.7
+          dist: xenial
+          sudo: true
```
Comment (iamareebjamal) [inline .travis.yml]: No need to use matrix. Just write dist: xenial

**Comment by iamareebjamal** [comment]
You have to add postgresql as service as well

https://docs.travis-ci.com/user/database-setup/#postgresql

**Pull request merged**